### DISCUSSION TITLE
Proposal for Developing a Chat GPT Plugin for SuttaCentral: Seeking Guidance and Feedback

### DISCUSSION THREAD

**Discussion by jgreen01** [suttacentral/suttacentral#2703] [Ideas]
Hello,

I am Jon Green, a senior web developer, and I am part of a small team working on a project to develop a Chat GPT plugin for SuttaCentral. Our team includes GitHub users harishvs, EqualAPriori, and @michaelh-sc . We are currently discussing our plans and progress in this thread on SuttaCentral: https://discourse.suttacentral.net/t/exploring-the-development-of-a-chatgpt-plugin-for-suttacentral/29404

Our goal is to create a plugin that will allow Chat GPT to interact with the SuttaCentral API, enabling users to ask questions and receive responses based on the early Buddhist texts available on SuttaCentral. We believe this will be a valuable tool for studying and engaging with these texts.

To achieve this, we need to create two main files: `ai-plugin.json` and `openapi.yaml` (or `openapi.json`). The `ai-plugin.json` file is the manifest file for the plugin, containing basic information about the plugin and the APIs it exposes to Chat GPT. The `openapi.yaml` (or `openapi.json`) file is the OpenAPI specification file for the API, defining a standard, language-agnostic interface to the HTTP APIs.

Our plan is to place these files in the `/.well-known/` directory of the domain, as specified in the OpenAI guidelines for creating a Chat GPT plugin. I believe these files can be added to the client webpack.prod.js, but I'm not 100% sure since this is my first time working with this code base.

Given that SuttaCentral uses Swagger, we should be able to generate an `openapi.yaml` file. However, we are not entirely sure about the process of obtaining the `swagger.json` file. We would appreciate any guidance or assistance you can provide on this matter.

Once we have the basic structure in place, we plan to start making pull requests to incorporate all the existing APIs. Following that, we will make PRs to add APIs that will enable traversing and searching the SuttaCentral database, among other functions. Does this seem like a reasonable approach?

We look forward to your feedback and any suggestions you might have.

Best,
Jon

**Comment by ozmos** [comment]
What a brilliant idea. I was just thinking it would be good to have the ability to "expand the pe..." when reading the Pali. This seems like a possible application of an AI assistant plugin.
